### PR TITLE
Test case for copycds, with a bogus disc

### PR DESCRIPTION
**Pull request by neo954**
This is for task https://github.com/xcat2/xcat2-task-management/issues/94
For issue #3254

### CODE REVIEW

**Comment by neo954** [comment]
```
# xcattest -t copycds_bogus_disc
xCAT automated test started at Tue May 22 04:44:37 2018
******************************
To detect current test environment
******************************
Detecting: OS = linux
Warning: No compute node defined, can't get ARCH of compute node
Warning: No compute node defined, can't get HCP TYPE of compute node
******************************
loading test cases
******************************
To run:
copycds_bogus_disc
******************************
Start to run test cases
******************************
------START::copycds_bogus_disc::Time:Tue May 22 04:44:38 2018------

RUN:/opt/xcat/share/xcat/tools/autotest/testcase/copycds/copycds_bogus_disc [Tue May 22 04:44:38 2018]
ElapsedTime:2 sec
RETURN rc = 0
OUTPUT:
+ trap cleanup EXIT
+ ISO_FILE=/opt/xcat/share/xcat/tools/autotest/testcase/copycds/BOGUS-RHEL-7.0-ppc64.iso
+ copycds /opt/xcat/share/xcat/tools/autotest/testcase/copycds/BOGUS-RHEL-7.0-ppc64.iso
Copying media to /install/rhels7/ppc64
Media copy operation successful
+ cleanup
+ rm -rf /install/rhels7/ppc64
+ rmdef -t osimage rhels7-ppc64-install-compute
1 object definitions have been removed.
+ rmdef -t osimage rhels7-ppc64-install-service
1 object definitions have been removed.
+ rmdef -t osimage rhels7-ppc64-netboot-compute
1 object definitions have been removed.
+ rmdef -t osimage rhels7-ppc64-statelite-compute
1 object definitions have been removed.
CHECK:rc == 0	[Pass]

------END::copycds_bogus_disc::Passed::Time:Tue May 22 04:44:40 2018 ::Duration::2 sec------
------Total: 1 , Failed: 0------

xCAT automated test finished at Tue May 22 04:44:40 2018
Please check results in the /opt/xcat/bin/../share/xcat/tools/autotest/result/,
and see /opt/xcat/bin/../share/xcat/tools/autotest/result//failedcases.20180522044437 file for failed cases.
see /opt/xcat/bin/../share/xcat/tools/autotest/result//performance.report.20180522044437 file for time consumption
```

File: xCAT-test/autotest/testcase/copycds/copycds_bogus_disc [hunk 1]
```diff
@@ -0,0 +1,18 @@
...
+	rm -rf /install/rhels7/ppc64
+	rmdef -t osimage rhels7-ppc64-install-compute
+	rmdef -t osimage rhels7-ppc64-install-service
+	rmdef -t osimage rhels7-ppc64-netboot-compute
+	rmdef -t osimage rhels7-ppc64-statelite-compute
+}
+
+trap cleanup EXIT
+
+ISO_FILE="/opt/xcat/share/xcat/tools/autotest/testcase/copycds/BOGUS-RHEL-7.0-ppc64.iso"
```
Comment (hu-weihua) [inline xCAT-test/autotest/testcase/copycds/copycds_bogus_disc]: Is ``BOGUS-RHEL-7.0-ppc64.iso`` a non-existed file? 
If it is true, how can you get return code 0?
```
[root@ copycds]# pwd
/opt/xcat/share/xcat/tools/autotest/testcase/copycds

[root@ copycds]# ll
total 8
-rw-r--r-- 1 root root 5195 Apr  3 06:16 cases0

[root@ copycds]# copycds  BOGUS-RHEL-7.0-ppc64.iso
Error: The management server was unable to find/read /opt/xcat/share/xcat/tools/autotest/testcase/copycds/BOGUS-RHEL-7.0-ppc64.iso. Ensure that file exists on the server at the specified location.
```

If ``BOGUS-RHEL-7.0-ppc64.iso`` is something should be prepared before running this case, it should be a attribute of this case.

File: xCAT-test/autotest/testcase/copycds/cases1 [hunk 2]
```diff
@@ -0,0 +1,5 @@
+start:copycds_bogus_disc
+description: Test cases for the copycds with a bogus disc
+cmd:/opt/xcat/share/xcat/tools/autotest/testcase/copycds/copycds_bogus_disc
```
Comment (hu-weihua) [inline xCAT-test/autotest/testcase/copycds/cases1]: If the code logic is sample and without attribute input, it is better not to use an additional scripts. 
Below code still work and it is convenient to debug when something goes wrong. 
```
cmd:#!/bin/bash
set -x

function cleanup()
{
	rm -rf /install/rhels7/ppc64
	rmdef -t osimage rhels7-ppc64-install-compute
	rmdef -t osimage rhels7-ppc64-install-service
	rmdef -t osimage rhels7-ppc64-netboot-compute
        rmdef -t osimage rhels7-ppc64-statelite-compute
}

trap cleanup EXIT

ISO_FILE="/opt/xcat/share/xcat/tools/autotest/testcase/copycds/BOGUS-RHEL-7.0-ppc64.iso"
check:rc==0
```

File: xCAT-test/autotest/testcase/copycds/copycds_bogus_disc (hunk 1 above)
Comment (neo954) [reply xCAT-test/autotest/testcase/copycds/copycds_bogus_disc]: File BOGUS-RHEL-7.0-ppc64.iso is included in this pull request.

File: xCAT-test/autotest/testcase/copycds/cases1 (hunk 2 above)
Comment (neo954) [reply xCAT-test/autotest/testcase/copycds/cases1]: The main design goal of the test case is not leave trash behind its running. Thus, do a proper cleanup before the test case exit is preferred.

This is a simple and small test case less than 20 lines. I believe it is far from being too complex to debugging. Actually, the effective code in this test case is only one line, `copycds BOGUS-RHEL-7.0-ppc64.iso`. When debugging, the debugger can run this command manually. Simple enter the line after bash prompt directly is good enough.

Comment (neo954) [reply xCAT-test/autotest/testcase/copycds/cases1]: I do not intend to use the `runscript` feature of `xcattest`. It is obfuscated and buggy.

Comment (neo954) [reply xCAT-test/autotest/testcase/copycds/cases1]: @hu-weihua, FYI, see #5254, #5255.

**Comment by hu-weihua** [comment]
@immarvin , do you have more comments about this case?

**Pull request merged**